com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -513,7 +518,7 @@ internal static void NetworkBehaviourUpdate(NetworkManager networkManager)
                             // Sync just the variables for just the objects this client sees
                             for (int k = 0; k < sobj.ChildNetworkBehaviours.Count; k++)
                             {
-                                sobj.ChildNetworkBehaviours[k].VariableUpdate(client.ClientId);
+                                sobj.ChildNetworkBehaviours[k].VariableUpdate(client.ClientId, k, snapshot);
```

Comment:
is `k` necessary? We already have `NetworkBehaviourId` property on NetworkBehaviour which could be used. I think right now it does not cache the behaviour id but since we don't allow it to change I'd change to `NetworkBehaviourId` to cache this and access it from there.

Reply:
Good point, let me try it, seems like I missed this accessor

Reply:
Sorry my initial message was a bit unclear. Now after this refactor we lost quite a bit of performance because `NetworkBehaviourId` is doing a look up each time. We should cache that as a private member field for the property.

Reply:
I missed this comment.  I did not realize this was a look-up. I'll add a task for it and make sure it goes in before next PR